src/project.c

```diff
@@ -790,16 +787,6 @@ void adjustpattern(int *pat, int index)
 	else if (*pat > index) (*pat)--;
 }
 
-
-void adjust_demand_pattern(list_node_t *list_node, int deletion_index)
-{
-	int pat_idx = get_pattern_index(list_node);
-
-	if (pat_idx == deletion_index) set_pattern_index(list_node, 0);
-	else if (pat_idx > deletion_index) set_pattern_index(list_node, --pat_idx);
-}
-
-
 void adjustpatterns(Network *network, int index)
```

Comment:
This function really makes the case why we should NOT be using indexes now that we are creating and deleting model elements.